src/orion/client/experiment.py

```diff
@@ -617,19 +622,26 @@ def workon(self, fct, max_trials=infinity, **kwargs):
         """
         self._check_if_executable()
 
-        trials = 0
-        kwargs = flatten(kwargs)
-        while not self.is_done and trials < max_trials:
-            trial = self.suggest()
-            if trial is None:
-                log.warning("Algorithm could not sample new points")
-                return trials
-            kwargs.update(flatten(trial.params))
-            results = fct(**unflatten(kwargs))
-            self.observe(trial, results=results)
-            trials += 1
-
-        return trials
+        with Parallel(n_jobs=n_jobs, verbose=100) as parallel:
+            trials = parallel(
+                delayed(self._optimize)(fct, kwargs) for _ in range(max_trials)
+            )
+
+        return sum(trials)
+
+    def _optimize(self, fct, *args, **kwargs):
+
+        # this is required for process-based or remote backend
+        setup_storage(storage=self.storage_config)
+
+        trial = self.suggest()
+        if trial is None:
+            log.warning("Algorithm could not sample new points")
+            return 0
+        kwargs.update(flatten(trial.params))
+        results = fct(**unflatten(kwargs))
+        self.observe(trial, results=results)
```

Comment:
Following comment above, turning this into a sequential optimization (basically moving code from `workon` to here, with small modifications)
```suggestion
        trials = 0
        trials_done = len(self.fetch_completed_trials())
        while not self.is_done and  len(self.fetch_completed_trials()) - trials_done < max_trials:
            trial = self.suggest()
            if trial is None:
                log.warning("Algorithm could not sample new points")
                return trials
            kwargs.update(flatten(trial.params))
            results = fct(**unflatten(kwargs))
            self.observe(trial, results=results)
            trials += 1
        return trials
```